File: atcoder.jp/PyPy3 (7.3.0)/abc285/abc285_b.py
N = int(input())
S = list(input())

for i in range(1, N):
    ans = 0
    for l in range(1, N - i + 1):
        k = 0
        while k < l and S[k] != S[k + i]:
            k += 1
        ans = max(ans, k)
    print(ans)

File: atcoder.jp/Rust (1.42.0)/abc285/abc285_b.rs
use proconio::{input};

fn main() {
    input! {
        n: usize,
        s: String,
    }

    let s = s.chars().collect::<Vec<_>>();

    for i in 1..n {
        let mut ans = 0;
        'outer: for l in 1..=n - i {
            for k in 1..=l {
                if s[k - 1] == s[k + i - 1] {
                    continue 'outer;
                }
            }
            ans = std::cmp::max(ans, l);
        }
        println!("{}", ans);
    }
}
